src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java

```diff
@@ -62,5 +75,79 @@ public void apply(Network network, Reporter subReporter) {
                 }
             }
         });
+
+        // apply transformers modifications
+        voltageInitModificationInfos.getTransformers().forEach(t -> {
+            if (t.getRatioTapChangerPosition() == null) {
+                return;
+            }
+
+            if (t.getLegSide() != null) {
+                ThreeWindingsTransformer threeWindingsTransformer = network.getThreeWindingsTransformer(t.getTransformerId());
+                if (threeWindingsTransformer == null) {
+                    Reporter genReporter = subReporter.createSubReporter(THREE_WINDINGS_TRANSFORMER_MSG + t.getTransformerId(), THREE_WINDINGS_TRANSFORMER_MSG + t.getTransformerId());
+                    genReporter.report(Report.builder()
+                        .withKey("3WindingsTransformerNotFound")
+                        .withDefaultMessage("3 windings transformer with id=${id} not found")
+                        .withValue("id", t.getTransformerId())
+                        .withSeverity(TypedValue.WARN_SEVERITY)
+                        .build());
+                    return;
+                }
+                if (threeWindingsTransformer.getLeg(t.getLegSide()).getRatioTapChanger() == null) {
```

Comment:
I wonder if we could have a NPE if getLeg() returns null.

Reply:
It's impossible : a three windings transformer has necessarly three legs ...